### ISSUE TITLE
Add options to notify player when all missions are sent

### ISSUE THREAD

**Issue by FREEZEball** [teelolws/TLDRMissions#189]
TLDR version: 
Add options to notify player when all missions are sent.

**Slightly longer version**
Problem: there is not enough indication that all missions are calculated and sent, so that player can close the table and log out/move on.
Some options might be:
1. Little toast sound is played.
2. [FlashClientIcon()](https://wowpedia.fandom.com/wiki/API_FlashClientIcon) if the game is in background.
3. A notification inside the TLDR window itself. There is a text string that changes to "all missions are sent" below the "Calculate" button, but it is barely visible (at least for me - the usecase is I start calculating missions and look at another monitor, and when everything is done, there is nothing to grab my attention).

**Comment by teelolws** [comment]
Hmm. I'll do some browsing of Blizzards sounds tomorrow. If I play a sound I'd prefer something not too intrusive.

**Comment by FREEZEball** [comment]
> If I play a sound I'd prefer something not too intrusive.

If player checks mark for a sound notification, they agree to hear that sound.

And, please, don't forget about those of us who play with sound turned off 😁.

**Comment by Kanegasi** [comment]
I suggest an input box for a sound ID or file ID.

Also, to expand on the last reply, either use the "Master" channel or an option to choose the channel.

**Comment by L30m4nc3r** [comment]
Maybe in this case there is still an opportunity to add an audio alert that some of the tasks have been completed? Let's say I'm playing on the main character and I see that more than four tasks were performed on my alternative characters. I mean to set some kind of limit of at least four completed missions for notification.

It will be nice to know that it's time to switch to other characters and resend missions.

**Comment by teelolws** [comment]
Okay heres what I'll do.

If you have WeakAuras installed, it'll send the custom event `TLDRMISSIONS_SENT_SUCCESS`. So you can now create a WeakAura that "listens" for that event, then shows whatever you want, plays whatever sound you want, whatever.

Heres one I quickly whipped up which plays a kaching sound:

    !WA:2!1r1tVnoru8gLdizHw6Mdb2chSar1IqQslROWfKACwhTbL0eSDPfPi5DSNNThQZmJMzCBZEcfHeCL(riN5u(eG6boBvXNGLVb7NaEJDaHqSZf)EJF)93V3B6mQ3YE0E0F(JvIktjJdQzNfnz8P(bPIsH6R7Ih5NszAzjzveCJjotOwsmXYydBjetxXjlzPXMcfOleL0vQvZYY0G55NCYF(l7HNTeEAHqnxW4MKH(Ng5hiDFZrSvRJC)0kTrS0AXzskXabWva38ksfElXWspNrnfEdqvfj1WeC9XbAdrzomqlQ4u5xnMBavgjfwmGsX)V4CGC5GkfrVyU4Aq1iofOmYIqRh6fxssly88Je55RPI4M48IKmgNPloS(DQ3V(HfEOMXzTrXYZbL(bhQ2j(7EMvsiPTUFLUkPPGdRYYy3SnE4GWO4WObbr1o3nSGyCNcAnjh(hlNRa0YGW5(tMSHcjOJryavp3FY8rNnXRcZBIf1avRdA5JIM8SGPJddhp70W4qeCJdpB4q)WWaKxaTZ22Qj2wz1oR1sOSCmv7u)UYdYQ4n42J)exfyQuCxJQcCbo9V9QGrHalRqVTIVRlDKp0c3xbrT6tfu4x3t(()VmARq9JFnZsfCs53IigMsy91cf9CfrU(8DcEzcUrU)if7LUFtfHQigIBu0geDcGCl9Eq)6(3JQjK0lZvwQXRuqOJsmKsen6hSSQ0WC8WMmTF9N7eKws0ARKNM9sWkS2MKqu5r3RliuX1x0oS2DJQjhw82ZGTWMubNYAMQCKF0BEAvQGuMTJ6ElHZSZLc(X1py0DardHgfWZnfVvDFpo203tXboRfn0Hsd2KOxBn1M46bEljm(O6tqhQhu7vpS(z43E)3Bu3SBf73O)0b)aUIP(EKVyzR8M4pkAJrKEvlm)EDEXfFiJExZuctBVsVRX)U2q8b7TntHtkyPI4D3x3URoYELTKschg47FAcwOzSCNpRIr3EdWRwo54NKs)Y7XLr08zYwGA7)cN65UTnrdBEhPtNoD3Gt3aTzT9hF7TmElsIE6CRgkZAFGWBwu0SP90jp9OV4ON80Ex9hx8xd

**Comment by zeenk** [comment]
Nice 👍

**Comment by Kanegasi** [comment]
Genius! It never occurred to me for addons to utilize the scanevents function from weakauras.

**Comment by teelolws** [comment]
Front page updated:

Addon Events for Weakauras:

If you want to hear a sound or flash the taskbar or whatever when the addon finishes calculations, you can set something up with Weakauras, and listen for one of these custom events that TLDRMissions will sent through AceEvent and directly to Weakauras if you have it installed:

    TLDRMISSIONS_SENT_SUCCESS (fired when all missions have been sent successfully)
    TLDRMISSIONS_NOT_ENOUGH_ANIMA (fired if the addon tried to send a mission but you didn't have enough Anima)
    TLDRMISSIONS_SENT_FAILURE (fired when the last possible mission failed, meaning you have followers left over not sent on any missions)
    TLDRMISSIONS_COMPLETE_MISSIONS_FINISHED (fired after you use the Finish Completed Missions button, and the addon has finished cycling through all the missions and collected their rewards)

**Comment by teelolws** [comment]
It might be a while before you can get the update. Curseforge is holding up my patch for some reason. I might just host it here for the weekend. 

![image](https://user-images.githubusercontent.com/49792789/168396887-12dbc2f8-4c90-46c4-a388-1c68c359b57e.png)

**Comment by FREEZEball** [comment]
Perfectly flexible!
